### PR TITLE
load route annotations from a directory

### PR DESCRIPTION
**Pull request by xabbuh**

### CODE REVIEW

**Comment by xabbuh** [comment]
In another PR it was suggested to parse directories recursively. I think we rather should not do that but simply handle only the files from the configured directory. What do you think?

File: Tests/Routing/Loader/DirectoryRouteLoaderTest.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    public function testLoad()
+    {
+        $collection = $this->loadFromDirectory(__DIR__.'/../../Fixtures/Controller');
+
+        $this->assertInstanceOf('Symfony\Component\Routing\Route', $collection->get('lock_user'));
+        $this->assertInstanceOf('Symfony\Component\Routing\Route', $collection->get('ban_user'));
+        $this->assertInstanceOf('Symfony\Component\Routing\Route', $collection->get('check_username_users'));
+        $this->assertInstanceOf('Symfony\Component\Routing\Route', $collection->get('new_users'));
+        $this->assertInstanceOf('Symfony\Component\Routing\Route', $collection->get('edit_user'));
+        $this->assertInstanceOf('Symfony\Component\Routing\Route', $collection->get('remove_user'));
```
Comment (xabbuh) [inline Tests/Routing/Loader/DirectoryRouteLoaderTest.php]: I am not really happy with this test. I think we should find a better way to ensure that the class does what it should.

**Comment by lsmith77** [comment]
we have an issue with 2.3:
```
1) FOS\RestBundle\Tests\Routing\Loader\DirectoryRouteLoaderTest::testLoad
BadMethodCallException: Unknown property "condition" on annotation "FOS\RestBundle\Controller\Annotations\Get".
```

**Comment by xabbuh** [comment]
Hm, looks like we cannot load the whole controller fixtures directory. So that's another reason to rearrange the tests.

**Comment by xabbuh** [comment]
Tests are fixed.

**Comment by lsmith77** [comment]
looks good to me

Status: Reviewed

**Comment by lsmith77** [comment]
can you open a PR for master to move this feature there?

**Comment by xabbuh** [comment]
will do

File: Routing/Loader/DirectoryRouteLoader.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+     */
+    public function load($resource, $type = null)
+    {
+        if (!is_dir($resource)) {
+            throw new \InvalidArgumentException(sprintf('Given resource of type "%s" is no directory.', $resource));
+        }
+
+        $collection = new RouteCollection();
+
+        $directoryIterator = new \DirectoryIterator($resource);
```
Comment (Ener-Getick) [inline Routing/Loader/DirectoryRouteLoader.php]: Why not using the symfony finders here ?

File: Routing/Loader/RestRouteLoader.php [hunk 3]
```diff
@@ -162,37 +162,8 @@ private function getControllerLocator($controller)
...
-
-            if (T_CLASS === $token[0]) {
-                $class = true;
-            }
-
-            if (T_NAMESPACE === $token[0]) {
-                $namespace = true;
-            }
-        }
+        @trigger_error('The '.__METHOD__.' method is deprecated since version 1.8 and will be removed in 2.0. Use '.__NAMESPACE__.'\ClassUtils::findClassInFile() instead.', E_USER_DEPRECATED);
```
Comment (Ener-Getick) [inline Routing/Loader/RestRouteLoader.php]: The last part of the message should be removed as the ``ClassUtils`` are ``@internal``

File: Tests/Routing/Loader/DirectoryRouteLoaderTest.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+class DirectoryRouteLoaderTest extends LoaderTest
+{
+    public function testLoad()
+    {
+        $collection = $this->loadFromDirectory(__DIR__.'/../../Fixtures/Controller/Directory');
+
+        $this->assertCount(9, $collection);
+
+        $this->assertInstanceOf('Symfony\Component\Routing\Route', $collection->get('get_users'));
```
Comment (Ener-Getick) [inline Tests/Routing/Loader/DirectoryRouteLoaderTest.php]: maybe replace this by:
```php
foreach ($collection->all() as $route) {
    $this->assertInstanceOf('Symfony\Component\Routing\Route', $route);
}
```
?

**Pull request merged**